### ISSUE TITLE
[refresh] navigation breaks on mobile when moving focus

### ISSUE THREAD

**Issue by alexgibson** [mozilla/bedrock#15154]
## Description

On mobile, tabbing through focus-able links sequentially breaks the layout.

### Steps to reproduce

1. In Firefox, open https://www-dev.allizom.org/en-US/
2. Using responsive design mode in dev tools, resize the browser down to 760px wide.
3. Using the keyboard, tab to the "Menu" link and press enter to open the menu.
4. Continue tabbing through the navigation links.
5. Observe that the layout breaks visually

![image](https://github.com/user-attachments/assets/ba32f07e-0b15-4cc4-8b06-0be353865646)

### Expected result

1. The layout should not break.
2. Focus should be retained inside the navigation menu (since it appears like a modal). Tabbing out to the last menu link should return focus back to the close button.

### Actual result

Right now focus is not restricted to the visual content. You can tab through to menus that are not expanded or visible to the user, which breaks the layout. This is likely an issue for screen reader users on mobile devices.

### Environment

Firefox
macOS
